telegram/functions.php

```diff
@@ -373,7 +373,7 @@ function editMessageText($id_val, $text_val, $markup_val, $chat_id = NULL, $merg
         $response = array_merge_recursive($response, $merge_args);
     }
 
-    if(!is_valid_target($chat_id, $id_val, true, false)){
+    if(!isset($response['inline_message_id']) && !is_valid_target($chat_id, $id_val, true, false)){
```

Comment:
Would it make more sense to add the check into is_valid_target and pass in more data? That way the logic is centralized instead of checking for two kinds of validity in every function.

Reply:
True, I'll edit it.